aea/manager/project.py

```diff
@@ -0,0 +1,246 @@
+        self.path: str = path
+        self.agents: Set[str] = set()
+
+    @classmethod
+    def load(
+        cls,
+        working_dir: str,
+        public_id: PublicId,
+        is_local: bool = False,
+        is_restore: bool = False,
+        registry_path: str = DEFAULT_REGISTRY_NAME,
+        skip_consistency_check: bool = False,
+    ) -> "Project":
+        """
+        Load project with given public_id to working_dir.
+
+        :param working_dir: the working directory
+        :param public_id: the public id
+        :param is_local: whether to fetch from local or remote
+        :param registry_path: the path to the registry locally
+        :param skip_consistency_check: consistency checks flag
+        """
+        ctx = Context(cwd=working_dir, registry_path=registry_path)
+        ctx.set_config("skip_consistency_check", skip_consistency_check)
+
+        path = os.path.join(working_dir, public_id.author, public_id.name)
+        target_dir = os.path.join(public_id.author, public_id.name)
+
+        if not is_restore and not os.path.exists(target_dir):
+            if is_local:
+                ctx.set_config("is_local", True)
+                fetch_agent_locally(ctx, public_id, target_dir=target_dir)
+            else:
+                fetch_agent(ctx, public_id, target_dir=target_dir)  # pragma: nocover
+
+        return cls(public_id, path)
+
+    def remove(self) -> None:
+        """Remove project, do cleanup."""
+        rmtree(self.path)
+
+    @property
+    def builder(self) -> AEABuilder:
+        """Get builder instance."""
+        return self._get_builder(self._get_agent_config(self.path), self.path)
+
+
+class AgentAlias(_Base):
+    """Agent alias representation."""
+
+    def __init__(
+        self, project: Project, agent_name: str, keys_dir: Optional[str],
+    ):
+        """Init agent alias with project, config, name, agent, builder."""
+        self.project = project
+        self.agent_name = agent_name
+        self._keys_dir = keys_dir or project.path
+        self._agent_config: AgentConfig = self._get_agent_config(project.path)
+
+    def set_agent_config_from_data(self, json_data: List[Dict]) -> None:
+        """
+        Set agent config instance constructed from json data.
+
+        :param json_data: agent config json data
+
+        :return: None
+        """
+        self._agent_config = AEABuilder.loader.load_agent_config_from_json(json_data)
+
+    @property
+    def builder(self) -> AEABuilder:
+        """Get builder instance."""
+        builder = self._get_builder(self.agent_config, self.project.path)
+        if not builder.private_key_paths:
+            # no keys, but create one for builder only
+            default_ledger = builder.get_default_ledger()
+            builder.add_private_key(
+                default_ledger, self._create_private_key(default_ledger)
+            )
+        builder.set_name(self.agent_name)
+        builder.set_runtime_mode("threaded")
```

Comment:
constantize

Reply:
this mode is not an option, this is one and the only supported and not supposed to be changed